### PR TITLE
Remove unnecessary conditional defines from corebluetooth.cs

### PR DESCRIPTION
**Pull request by chamons**
- Unlike the rest of my recent work, this is all done by hand, so please review closely.
- Part of https://github.com/xamarin/xamarin-macios/issues/14444

One of the define blocks included an API mistake work around, so it seemed easier to do it this way. Please check my work doubly.

### CODE REVIEW

File: src/corebluetooth.cs [hunk 1]
```diff
@@ -763,21 +763,25 @@ interface CBUUID : NSCopying {
...
-#if MONOMAC
 		[Internal]
 		[Field ("CBUUIDValidRangeString")]
 		[Introduced (PlatformName.MacOSX, 10, 12)]
 		[Deprecated (PlatformName.MacOSX, 10, 13)]
 		[Obsoleted (PlatformName.MacOSX, 10, 13)]
+		[NoiOS][NoTV][NoWatch][NoMacCatalyst]
 		NSString CBUUIDValidRangeString { get; }
 
+#if MONOMAC && !NET
```
Comment (chamons) [inline src/corebluetooth.cs:774]: This is the original binding mistakes. I made it legacy, and then made the correct binding all platforms on NET

File: src/corebluetooth.cs [hunk 2]
```diff
@@ -850,13 +854,9 @@ interface CBATTRequest {
 	// `delloc` a default instance crash applications and a default instance, without the ability to change the UUID, does not make sense
 	[DisableDefaultCtor]
 	interface CBCentral : NSCopying {
-#if MONOMAC
-		// Introduced with iOS7, but does not have NS_AVAILABLE
-		// Moved to a new base class, CBPeer, in iOS 8.
-		[iOS (7,0)]
```
Comment (chamons) [inline src/corebluetooth.cs:856]: This iOS seems to have been a copy & paste issue, as we were in a MONOMAC block.

**Review by rolfbjarne — APPROVED** [review]

File: src/corebluetooth.cs [hunk 3]
```diff
@@ -50,23 +49,24 @@ interface CBManager {
...
 		[Internal]
 		[iOS (13,1), Watch (6,1)]
+		[NoTV][NoMac]
 		[Static]
 		[Export ("authorization", ArgumentSemantic.Assign)]
 		CBManagerAuthorization _SAuthorization { get; }
-#else
+
 		[TV (13,0), Mac (10, 15)]
+		[NoiOS][NoWatch]
```
Comment (tj-devel709) [inline src/corebluetooth.cs:66]: Does this need a [NoMacCatalyst]?

File: src/corebluetooth.cs [hunk 4]
```diff
@@ -763,21 +763,25 @@ interface CBUUID : NSCopying {
...
 		[Internal]
 		[Mac (10,13)]
 		[Field ("CBUUIDCharacteristicValidRangeString")]
 		NSString CBUUIDCharacteristicValidRangeString { get; }
 #else
 		[iOS (10,0)]
 		[TV (10,0)]
+		[Watch (4,0)]
+		[Mac (10,13)]
+		[MacCatalyst (13,0)]
```
Comment (tj-devel709) [inline src/corebluetooth.cs:784]: Also it looks like the min MacCatalyst version we support is 13.1, I don't think it should matter too much though right?

**Review by mandel-macaque — APPROVED** [review]

**Comment by mandel-macaque** [comment]
Landing since the watch failure is due to a network issue: But was:  BadGateway - FAIL] GHIssue8344() :   Second status not ok

**Pull request merged**